app/services/user_opportunity_discovery.py

```
@@ -4467,6 +4553,50 @@ async def _track_scan_lifecycle(self, user_id: str, scan_id: str, phase: str, st
         except Exception as track_error:
             self.logger.debug("Lifecycle tracking failed", error=str(track_error))
 
+    async def _track_debug_step(self, user_id: str, scan_id: str, step_number: int, step_name: str, status: str = "in_progress", error: str = None, **extra_data):
```

Ruff (0.14.1)

Unused method argument: `user_id`

(ARG002)